lib/tapioca/core_ext/class.rb

```diff
@@ -15,9 +17,10 @@ class Class
   #
   #   class D < C; end
   #   C.descendants # => [B, A, D]
+  sig { returns(T::Array[Class]) }
   def descendants
-    ObjectSpace.each_object(singleton_class).reject do |k|
-      k.singleton_class? || k == self
-    end
+    T.cast(ObjectSpace.each_object(singleton_class).reject do |k|
+      T.cast(k, Module).singleton_class? || k == self
+    end, T::Array[Class])
```

Comment:
Wdyt about using a local variable so the cast is not split over 3 lines?

```suggestion
    res = ObjectSpace.each_object(singleton_class).reject do |k|
      T.cast(k, Module).singleton_class? || k == self
    end
    T.cast(res, T::Array[Class])
```